### DISCUSSION TITLE
Use PageNumberWithinSection and TotalPagesWithinSection in Dynamic components

### DISCUSSION THREAD

**Discussion by simusr2** [QuestPDF/QuestPDF#469] [Q&A] [answered]
Is possibile to write blocks with conditions based on the section page numbers instead of global document numbers?

I'm using Dynamic components to show if the current page is the first, the last or a "inner" one.

In the following code is there a method to use  context.PageNumberWithinSection and context.TotalPagesWithinSection?

Thanks in advice!

```cs
public class CurrentTextPage : IDynamicComponent<int>
{
    private string _currentSection;
    public CurrentTextPage(string currentSection)
    {
        _currentSection = currentSection;
    }

    public int State { get; set; }

    public DynamicComponentComposeResult Compose(DynamicContext context)
    {
        IDynamicElement content = context.CreateElement(container =>
        {
            int currentPage = context.PageNumber;
            int totalPages = context.TotalPages;

            if (currentPage == 1)
            {
                container.Text("First");
            }
            else if (currentPage == totalPages)
            {
                container.Text("Last");
            }
            else
            {
                container.Text("In");
            }
        });

        return new DynamicComponentComposeResult
        {
            Content = content,
            HasMoreContent = false
        };
    }
}
```

**Comment by skitchpatterson3** [comment]
This might help me, too. I'm trying to find a way to customize the page numbers appearing in my Table of Contents so that I can display the entry's page number _within its section_ instead of the page number of the overall document. It's common to make "page 1" of a document as the first page containing the main content, skipping any cover pages and TOC pages. But I can't seem to figure out how to pull that off yet.

**Reply by simusr2** [reply]
Hi, I made the PR #482 for this topic. It was for an old version of QuestPDF but you can check my implementation in the PR.

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Per-section page numbering is available through `PageNumberWithinSection()` and `TotalPagesWithinSection()` in text elements. You can find the documentation here:
https://www.questpdf.com/api-reference/section.html
https://www.questpdf.com/api-reference/text/page-numbers.html

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
